File: app/types/export.ts
/**
 * エクスポート形式
 * Rust側の ExportFormat enum と対応
 * serde(rename_all = "camelCase") により "csv", "excel", "json" にシリアライズされる
 */
export type ExportFormatType = 'csv' | 'excel' | 'json'

export interface ExportOptions {
    path: string
    format: ExportFormatType
}

export interface ExportResult {
    success: boolean
    message: string | null
    rowsAffected: number
}

File: src-tauri/src/models/export.rs
use serde::{Deserialize, Serialize};
use std::path::PathBuf;

#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
#[serde(rename_all = "camelCase")]
pub enum ExportFormat {
    Csv,
    Excel,
    Json,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ExportOptions {
    pub path: PathBuf,
    pub format: ExportFormat,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ExportResult {
    pub success: bool,
    pub message: Option<String>,
    pub rows_affected: usize,
}
